main.go

```diff
@@ -37,11 +37,25 @@ func main() {
 						error := errorConstructor.New(err.Error())
 						reject.Invoke(error)
 					} else {
-						var stmts []interface{}
-						for _, stmt := range ast.Statements {
-							stmts = append(stmts, stmt.String())
+						response := make(map[string]interface{})
+						statements := make([]interface{}, len(ast.Statements))
+						var typ string
+						for i, stmt := range ast.Statements {
+							if typ == "" {
```

Comment:
if we don't already have a type, check it. for create and read... we'll only ever have one statement here, and the parser will catch and guarantee that for us. but for write statements, we don't need to check more than once.